### PR TITLE
First version

### PR DESCRIPTION
**Pull request by pa-cholek**

### CODE REVIEW

**Review by qunabu — APPROVED** [review]

**Review by qunabu — APPROVED** [review]
tests are failing

**Comment by pa-cholek** [comment]
@qunabu This is just a part of larger codebase, working tests are in part 3/3

File: database/migrations/2022_01_27_000000_create_coupons_table.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            $table->softDeletes();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
```
Comment (KrzysztofDziedziechEscolasoft) [inline database/migrations/2022_01_27_000000_create_coupons_table.php:49]: public function down(): void

File: database/migrations/2022_01_27_000000_create_coupons_table.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+            $table->unsignedInteger('amount');
+
+            $table->timestamps();
+            $table->softDeletes();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
```
Comment (KrzysztofDziedziechEscolasoft) [inline database/migrations/2022_01_27_000000_create_coupons_table.php:45]: remove

**Review by KrzysztofDziedziechEscolasoft — APPROVED** [review]

**Comment by pa-cholek** [comment]
I've merged parts 2/3 and 3/3 to this branch, so this has everything now

**Pull request merged**